src/backend/d3d12/QueueD3D12.cpp

```diff
@@ -31,18 +31,29 @@ namespace d3d12 {
             nullptr,
             IID_PPV_ARGS(&commandList)
         ));
-        ASSERT_SUCCESS(commandList->Close());
 
         ASSERT_SUCCESS(device->GetD3D12Device()->CreateFence(fenceValue, D3D12_FENCE_FLAG_NONE, IID_PPV_ARGS(&fence)));
         fenceEvent = CreateEvent(nullptr, FALSE, FALSE, nullptr);
         ASSERT(fenceEvent != nullptr);
     }
 
     void Queue::Submit(uint32_t numCommands, CommandBuffer* const * commands) {
-        // TODO(enga@google.com): This will stall on the previous submit because
+        ComPtr<ID3D12CommandAllocator> pendingCommandAllocator = device->GetPendingCommandAllocator();
```

Comment:
LGTM, but this will need to be revisited to not stall and properly track objects that are in flight.